runtime/lua/vim/diagnostic.lua

```diff
@@ -976,19 +976,19 @@ function M._get_virt_text_chunks(line_diags, opts)
   -- Create a little more space between virtual text and contents
   local virt_texts = {{string.rep(" ", spacing)}}
 
-  for i = 1, #line_diags - 1 do
+  for i = 2, #line_diags do
```

Comment:
Suggestion: reverse the order of insertion so that the little squares are ordered (from left to right) `N, N-1, ..., 1` instead of `2, 3, ..., N, 1`. This creates the visual appearance of a "stack", which is nice, and what we have today, but with the order reversed so that the first diagnostic is visible rather than the last.

```suggestion
  for i = #line_diags, 2, -1 do
```

Reply:
Isn't it kinda weird though that diagnostics may go left to right but the stack goes right to left?

Reply:
Hmm, maybe. If that's confusing then maybe we still order the prefix squares from left to right in the original order. I just find the `2, 3, ..., N, 1` order nonintuitive.

Reply:
It's weird because I find the order unintuitive, but the virtual text more intuitive (first error first). Maybe we should drop the prefix or make it a number instead? Idk